src/new.rs

```diff
@@ -613,3 +630,21 @@ fn validate_name(name: &str) -> Result<(), Error> {
         Ok(())
     }
 }
+
+fn get_foldername(path: &str) -> Option<String> {
+    match path {
+        "." => Some(
+            env::current_dir()
+                .gleam_expect("invalid folder")
+                .file_name()
+                .and_then(|x| x.to_str())
+                .map(ToString::to_string)?,
+        ),
+        _ => Some(
+            Path::new(path)
+                .file_name()
+                .and_then(|x| x.to_str())
+                .map(ToString::to_string)?,
+        ),
+    }
```

Comment:
No need to use the `?` operator and then wrap result into `Some`, `map` already returns an Option